### PR TITLE
Resolves #22 and #20 More documentation + added most mdx components

### LINKED ISSUE #20: Do Github Slugs for Markdown rendering
https://github.com/Flet/github-slugger

### LINKED ISSUE #22: Create Components for MDX
mdx components that map to elements in markdown

### PR DESCRIPTION
**Pull request by tko22**
Resolves #22 
Resolves #20
- add sections to documentation
- added slugs to headings, so you can link sections within a page
- added most mdx components (not tables will do that once tables are done)
- added border to Box
- created Text and Heading documentation pages

## sections
image

## Linking markdown titles
image

### CODE REVIEW

File: docs/components/mdx-components.js [hunk 1]
```diff
@@ -27,9 +43,87 @@ const InlineCode = ({ children }) => (
...
-  h1: (props) => <Heading type="h1" {...props} />,
-  h2: (props) => <Heading type="h2" {...props} />,
+  p: (props) => <Text as="p" type="body" {...props} />,
+  h1: (props) => <PermalinkHeading type="h1" {...props} />,
+  h2: (props) => <PermalinkHeading type="h2" {...props} />,
+  h3: (props) => <PermalinkHeading type="h3" {...props} />,
+  h4: (props) => <PermalinkHeading type="h4" {...props} />,
+  // don't have styling for h5 or h6, use h4
+  h5: (props) => <PermalinkHeading type="h1" {...props} />,
+  h6: (props) => <PermalinkHeading type="h1" {...props} />,
```
Comment (jackieo5023) [inline docs/components/mdx-components.js:123]: is this supposed to be `type="h4"` or am i misunderstanding?

File: docs/pages/components/grid.mdx [hunk 2]
```diff
@@ -0,0 +1,9 @@
+export const meta = {type: "components", title: "Grid"}
+
+import { Grid } from '@hack4impact-uiuc/bridge'
+import CodeExampleBox from '../../components/CodeExampleBox'
+
+import Layout from '../../components/layout.js'
+export default Layout({...meta})
+
+gird
```
Comment (jackieo5023) [inline docs/pages/components/grid.mdx:9]: nice

**Comment by jackieo5023** [comment]
WHOOPS i just realized the tests failed

File: docs/components/mdx-components.js (hunk 1 above)
Comment (tko22) [reply docs/components/mdx-components.js:123]: haha yes ill fix that

File: docs/pages/components/grid.mdx (hunk 2 above)
Comment (tko22) [reply docs/pages/components/grid.mdx:9]: LOL WOOPS

**Pull request merged**